### PR TITLE
pfetch: initial solaris support

### PR DESCRIPTION
**Pull request by Crestwave**
Note that this has only been tested on OpenIndiana Hipster.

### CODE REVIEW

File: pfetch [hunk 1]
```diff
@@ -246,6 +246,12 @@ get_os() {
             trap '' EXIT
         ;;
 
+        SunOS)
+            read -r distro branch version _ < /etc/release
```
Comment (dylanaraps) [inline pfetch]: Full output of `cat /etc/release`?

File: pfetch [hunk 2]
```diff
@@ -367,6 +373,14 @@ get_uptime() {
             # regular seconds.
             s=$(($(system_time) / 1000000))
         ;;
+
+        SunOS)
+            read -r _ s <<-EOF
+                $(kstat -p unix:0:system_misc:snaptime)
```
Comment (dylanaraps) [inline pfetch]: Full output of `kstat -p unix:0:system_misc:snaptime`?

File: pfetch [hunk 3]
```diff
@@ -612,6 +631,21 @@ get_memory() {
...
             mem_full=$(( mem_full / 1024))
         ;;
+
+        SunOS)
+            hw_pagesize=$(pagesize)
+
+            while read -r _ p; do
+                : "${pages_total:+"${pages_free="$p"}"}" "${pages_total="$p"}"
+            done <<-EOF
+				$(kstat -p unix:0:system_pages:pagestotal \
```
Comment (dylanaraps) [inline pfetch]: Full output of: 

```sh
kstat -p unix:0:system_pages:pagestotal unix:0:system_pages:pagesfree
```

File: pfetch (hunk 1 above)
Comment (Crestwave) [reply pfetch]: I was still figuring out how to send it. I guess I'll have to settle for uploading it to a Git repository

Comment (Crestwave) [reply pfetch]:
```
             OpenIndiana Hipster 2019.04 (powered by illumos)
        OpenIndiana Project, part of The Illumos Foundation (C) 2010-2019
                        Use is subject to license terms.
                           Assembled 10 May 2019
```

File: pfetch (hunk 3 above)
Comment (Crestwave) [reply pfetch]:
```
unix:0:system_pages:pagestotal	1046397
unix:0:system_pages:pagesfree	885018
```

File: pfetch (hunk 2 above)
Comment (Crestwave) [reply pfetch]:
```
unix:0:system_misc:snaptime	14809.906993005
```

File: pfetch (hunk 1 above)
Comment (dylanaraps) [reply pfetch]:
```sh
IFS='(' read -r distro _ < /etc/release
```

File: pfetch (hunk 3 above)
Comment (dylanaraps) [reply pfetch]: I think this is clearer: 

```sh
while read -r key val; do
    case $key in
        *total) mem_full=$val ;;
        *free)  mem_free=$val ;;
    esac
done <<-EOF
	$(kstat -p unix:0:system_pages:pagestotal \
	           unix:0:system_pages:pagesfree)
EOF
```

Comment (dylanaraps) [reply pfetch]: (Will fix indentation when it hits master)

File: pfetch [hunk 4]
```diff
@@ -456,6 +466,11 @@ get_pkgs() {
             Minix)
                 printf '%s\n' /usr/pkg/var/db/pkg/*/
             ;;
+
+            SunOS)
+                has pkginfo    && pkginfo -i
```
Comment (dylanaraps) [inline pfetch]:
```sh
                has pkginfo && pkginfo -i
                has pkg     && pkg list
```

File: pfetch (hunk 3 above)
Comment (Crestwave) [reply pfetch]: You mean `pages_total` and `pages_free`?

File: pfetch [hunk 5]
```diff
@@ -246,6 +246,13 @@ get_os() {
             trap '' EXIT
         ;;
 
+        SunOS)
+            IFS='(' read -r distro _ < /etc/release
+            kstat=$(kstat -p unix:0:system_misc:snaptime \
```
Comment (dylanaraps) [inline pfetch]: I was thinking more along the lines of adding the parser here *once*.

The `case` statement in the loop to populate `pages_free`, `pages_total`, and `seconds` (can be further stripped of `.` in `get_uptime`).

**Comment by dylanaraps** [comment]
LGTM :+1:

**Pull request merged**